src/fastmcp/server/context.py

```
@@ -1156,32 +1161,64 @@ def _make_state_key(self, key: str) -> str:
+                key=prefixed_key,
+                value=StateValue(value=value),
+                ttl=self._STATE_TTL_SECONDS,
+            )
+        except Exception as e:
+            if "serialize" in str(e).lower():
+                raise TypeError(
+                    f"Value for state key {key!r} is not serializable. "
+                    f"Use set_state({key!r}, value, serializable=False) to store "
+                    f"non-serializable values. Note: non-serializable state is "
+                    f"request-scoped and will not persist across requests."
+                ) from e
```

Ruff (0.15.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)